code/modules/lighting/lighting_mask/shadow_calculator.dm

```diff
@@ -0,0 +1,529 @@
+//#define SHADOW_DEBUG
+//Lighting texture scales in world units (divide by 32)
+//256 = 8,4,2
+//1024 = 32,16,8
+#define LIGHTING_SHADOW_TEX_SIZE 8
+
+#define COORD_LIST_ADD(listtoadd, x, y) \
+	if(islist(listtoadd["[x]"])) { \
+		var/list/_L = listtoadd["[x]"]; \
+		BINARY_INSERT_NUM(y, _L); \
+	} else { \
+		listtoadd["[x]"] = list(y);\
+	}
+
+#ifdef SHADOW_DEBUG
+#define DEBUG_HIGHLIGHT(x, y, colour) \
+	do { \
+		var/turf/T = locate(x, y, 2); \
+		if(T) { \
+			T.color = colour; \
+		}\
+	} while (0)
+#define DO_SOMETHING_IF_DEBUGGING_SHADOWS(something) something
+#else
+#define DEBUG_HIGHLIGHT(x, y, colour)
+#define DO_SOMETHING_IF_DEBUGGING_SHADOWS(something)
+#endif
+
+/atom/movable/lighting_mask
+	var/list/affecting_turfs
```

Comment:
should be a typed list

looks like `as()` is used with this directly below, but not in other iteration references.

unsure if `as()` should be used with this - do turfs that get replaced/deleted end up as null entries in the list? if so, as() would break

Reply:
Will update accordingly